src/Castle.Core/DynamicProxy/Internal/AttributeUtil.cs

```diff
@@ -243,6 +263,11 @@ private static bool ShouldSkipAttributeReplication(Type attribute)
 				return false;
 			}
 
+			if (ignoreInheritance)
```

Comment:
It might be best to wrap the actual inheritance check with `if (!ignoreInheritance)` instead in case another check gets added to this method since at the moment the ordering doesn't really matter.

Reply:
Done.